### PR TITLE
feat: homepage gallery component 

### LINKED ISSUE #1288: Name collisions in SVG `Image` import
**Describe the bug**

When there are more than one diagram on the page, imported images might not show up. This is usually caused by name collisions among the imported SVGs. If an imported SVG defines a `<linearGradient id="foo">` and refer to `foo` by `url(#foo)` somewhere else, there must be only one such reference in the entire DOM tree. 

**To Reproduce**

1. Go to https://361060b0.penrose-72l.pages.dev/try/
2. Click on the "Walk on Spheres" example
3. Select `Grid`
4. See the following output

image

**Expected behavior**

Spheres should be shaded by the imported image:

Walk on Spheres - Laplace Estimator

**Additional context**
This is a known problem with SVG (e.g. https://github.com/gregberge/svgr/issues/150) and there are some SVG cleaning tools like [svgr](https://github.com/gregberge/svgr). It would be great to experiment with some of them.

### PR DESCRIPTION
**Pull request by wodeni**
# Description

Resolves #1288. Resolves https://github.com/penrose/penrose/issues/1307.

This PR adds a new `Gallery` component to `components`, which is used in the "Examples" page on our site. To make this work, I had to fix a SVG and `Image` related issue: #1288. The overall strategy is to run `svgo.optimize` with two plugins ([`prefixIds`](https://github.com/svg/svgo/blob/main/plugins/prefixIds.js) and [`inlineStyles`](https://github.com/svg/svgo/blob/main/plugins/inlineStyles.js)) at two places: `yarn registry` test and `editor` SVG download.

# Implementation strategy and design decisions

* #1307: `inlineStyles` fixes this issue
* #1288: we are still using `makeIdsUnique` introduced in https://github.com/penrose/penrose/pull/1287 because running `svgo` in real time can be quite slow. In `Gallery`, since we are pulling from `main`'s cached SVGs, they are already optimized using `prefixIds`. 

# Examples with steps to reproduce them

# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have reviewed any generated registry diagram changes

# Open question

* In some cases, we might be using `prefixIds` and `makeIdsUnique` concurrently. This doesn't seem to cause a problem in general, but it's a bit odd.
* A lot of the diagrams don't look great in dark mode. I'd suggest turning it off in another PR.
* Currently both `Gallery` and `ExampleBrowser` pull SVGs from `main`. We should find a way to run the registry and copy the cached SVGs into `docs-site` in another PR.

### CODE REVIEW

**Comment by jiriminarcik** [comment]
> Thanks @wodeni! Is the idea that these examples will show up on the landing page (https://penrose.cs.cmu.edu/)?

You can preview the landing page here:
https://197bcd09.penrose-72l.pages.dev/
There is a new Examples tab.

File: packages/docs-site/package.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "@penrose/docs-site",
   "version": "3.0.0-alpha.1",
+  "type": "module",
```
Comment (samestep) [inline packages/docs-site/package.json:4]: based

**Review by samestep — APPROVED** [review]
Awesome work, thanks so much for doing this! The `new DOMParser()` and `new XMLSerializer()` stuff looks pretty intense.

File: packages/docs-site/src/components/GalleryWrapper.vue [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<script setup lang="ts">
+import { withBase } from "vitepress";
+import { defineAsyncComponent } from "vue";
+
+const Gallery = defineAsyncComponent(async () => {
+  const { applyPureReactInVue } = await import("veaury");
+  const { Gallery } = await import("@penrose/components");
+  return applyPureReactInVue(Gallery);
+});
```
[the remark this answers is not in the document]
Comment (wodeni) [reply packages/docs-site/src/components/GalleryWrapper.vue:9]: Guess so. Just copying over the usage from `Demo` and didn't really learn veaury. The docs are a little too intense for me.

File: packages/editor/src/state/callbacks.ts [hunk 3]
```diff
@@ -238,7 +239,9 @@ export const useLoadExampleWorkspace = () =>
         const id = toast.loading("Loading example...");
         const { domain, style, substance, variation } = await meta.get();
         toast.dismiss(id);
-        const styleJoined = style.map(({ contents }) => contents).join("\n");
+        const styleJoined = style
+          .map(({ contents }: any) => contents)
```
[the remark this answers is not in the document]
Comment (wodeni) [reply packages/editor/src/state/callbacks.ts]: Looks like I needed to import `registry` by doing `import registry from "@penrose/examples/dist/registry.js";` or the types will not resolve. Not really sure what's going on there. [9f2a21b](https://github.com/penrose/penrose/pull/1464/commits/9f2a21bf06f7af4688ee1725bb6aac006be0c0cb) fixed this issue.

**Pull request merged**